### PR TITLE
Enforce ordering and tighten up on LogBroadcaster sub/unsub

### PR DESCRIPTION
**Pull request by samsondav**

### CODE REVIEW

File: core/chains/evm/log/broadcaster.go [hunk 1]
```diff
@@ -204,21 +214,34 @@ func (b *broadcaster) awaitInitialSubscribers() {
...
-	if wasOverCapacity {
-		b.logger.Error("Subscription mailbox is over capacity - dropped the oldest unprocessed subscription")
-	}
-	return func() {
-		wasOverCapacity := b.rmSubscriber.Deliver(reg)
+		sub := &subscriber{listener, opts}
+		wasOverCapacity := b.changeSubscriberStatus.Deliver(changeSubscriberStatus{subscriberStatusSubscribe, sub})
 		if wasOverCapacity {
-			b.logger.Error("Subscription removal mailbox is over capacity - dropped the oldest unprocessed removal")
+			panic("LogBroadcaster subscribe: changeSubscriberStatus channel was full")
```
Comment (pinebit) [inline core/chains/evm/log/broadcaster.go:231]: For what reason you used `panic` instead of `b.logger.Panic`?

Comment (samsondav) [reply core/chains/evm/log/broadcaster.go:232]: No reason, I will fix.

File: core/chains/evm/log/registrations.go [hunk 2]
```diff
@@ -183,77 +196,79 @@ func filtersContainValues(topicValues []common.Hash, filters [][]Topic) bool {
...
 }
 
-func (r *subscribers) removeSubscriber(reg registration) (needsResubscribe bool) {
-	addr := reg.opts.Contract
+func (r *handlers) removeSubscriber(sub *subscriber) (needsResubscribe bool) {
+	addr := sub.opts.Contract
 
-	if _, exists := r.handlers[addr]; !exists {
+	// FIXME: What about the case where you remove/add a job with the same contract address?
+	// addr is not good enough to be a unique key
```
Comment (samsondav) [inline core/chains/evm/log/registrations.go:235]: I suspect this might be causing the bug where jobs can be removed/re-added but don't resubscribe

Comment (samsondav) [reply core/chains/evm/log/registrations.go:235]: I will fix in a followup PR

**Review by pinebit — APPROVED** [review]
LGTM

**Review by pinebit — APPROVED** [review]

**Pull request merged**